modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,712 @@
+                   " vertical/horizontal expansion triggered unneccessary redraw,"
+                   " PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+
+            DestroyWindow(Window);
+
+            FsmState = FSM_STATE_END;
+            break;
+
+        case FSM_STATE_END:
+            break;
+    }
+
+    if (hdc != NULL)
+    {
+        PrevColor = Color;
+        ReleaseDC(Window, hdc);
+    }
+
+    return 0;
+}
+
+static int OnResize(HWND Window, int NewWidth, int NewHeight)
+{
+    CurrentColor = (CurrentColor + 1) % TEST_COLOR_COUNT;
+    SetClassLongPtr(Window,
+                    GCLP_HBRBACKGROUND,
+                    (LONG_PTR)ColorBrushes[CurrentColor]);
+
+    trace("New window size: %d x %d, new color: 0x%.8lX\n",
+          NewWidth, NewHeight, Colors[CurrentColor]);
+
+    return 0;
+}
+
+static LRESULT CALLBACK WindowProc(HWND Window, UINT Message, WPARAM wParam, LPARAM lParam)
+{
+    RECT Rect;
+    BOOL BoolRC;
+    int Width;
+    int Height;
+
+    switch (Message)
+    {
+        case WM_CREATE:
+
+            /* It's important for the test that the entire Window is visible. */
+            BoolRC = SetWindowPos(Window, HWND_TOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE);
+            ok(BoolRC, "Failed to set window as top-most, code: %ld\n", GetLastError());
+
+            BoolRC = GetClientRect(Window, &Rect);
+            ok(BoolRC, "Failed to retrieve client area dimensions, code: %ld\n", GetLastError());
+
+            if (BoolRC)
+            {
+                ClientWidth = Rect.right;
+                ClientHeight = Rect.bottom;
+            }
+            else
+            {
+                return -1;
+            }
+
+            BoolRC = GetWindowRect(Window, &Rect);
+            ok(BoolRC, "Failed to retrieve window dimensions, code: %ld\n", GetLastError());
+
+            if (BoolRC)
+            {
+                OrigWidth  = Rect.right - Rect.left;
+                OrigHeight = Rect.bottom - Rect.top;
+            }
+            else
+            {
+                return -1;
+            }
+
+            SmallWidth  = max((OrigWidth  * 3) / 4, 1);
+            SmallHeight = max((OrigHeight * 3) / 4, 1);
+            OrigWidth   = max(OrigWidth,  SmallWidth+1);
+            OrigHeight  = max(OrigHeight, SmallHeight+1);
+
+            trace("OrigWidth: %d, OrigHeight: %d, SmallWidth: %d, SmallHeight: %d\n",
+                  OrigWidth, OrigHeight, SmallWidth, SmallHeight);
+
+            HideVertScrollBar(Window);
+            HideHorzScrollBar(Window);
+
+            FsmState = FSM_STATE_START;
+            FsmTimer = 0;
+
+            return 0;
+
+        case WM_PAINT:
+
+            if (FsmTimer == 0)
+            {
+                FsmTimer = SetTimer(Window, 1, FSM_STEP_PERIOD_MS, NULL);
+                ok(FsmTimer != 0, "Failed to initialize FSM timer, code: %ld\n", GetLastError());
+
```

Comment:
```suggestion
```

Reply:
doing a call, and then checking its retval is dealing with the same thing. Upvoting Hermes suggestion therefore.

Reply:
Whoops, you're right, I missed this one, sorry!